pyqt-apps/siriushla/as_rf_control/details.py

```diff
@@ -411,3 +413,234 @@ def _setupUi(self):
+        # Fifth line
+        self.sb_delay_sample = SiriusSpinbox(self,
+                                         self.prefix + self.chs['FDL']['Delay'] + '-SP')
+        self.lb_delay_sample = SiriusLabel(self,
+                                         self.prefix + self.chs['FDL']['Delay'] + '-RB')
+        self.sb_delay_us = SiriusSpinbox(self,
+                                         self.prefix + self.chs['FDL']['Delay'] + 'Time-SP')
+        self.lb_delay_us = SiriusLabel(self,
+                                         self.prefix + self.chs['FDL']['Delay'] + 'Time-RB')
+        sb_unit = QComboBox()
+        sb_unit.addItems(['Choose a unit', 'Sample units', 'us'])
+        sb_unit.setMaximumWidth(120)
+        sb_unit.currentTextChanged.connect(self._handle_unit_change)
+
+        delay_lay = QHBoxLayout()
+        delay_lay.addWidget(self.sb_delay_sample)
+        delay_lay.addWidget(self.lb_delay_sample)
+        delay_lay.addWidget(self.sb_delay_us)
+        delay_lay.addWidget(self.lb_delay_us)
+
+        self.sb_delay_us.setVisible(False)
+        self.lb_delay_us.setVisible(False)
+        self.sb_delay_sample.setVisible(False)
+        self.lb_delay_sample.setVisible(False)
+
+        lay.addWidget(QLabel(
+            '<h4>Trigger Delay:</h4>', self, alignment=Qt.AlignRight | Qt.AlignVCenter),
+            4, 0
+        )
+        lay.addWidget(sb_unit, 4, 1)
+        lay.addLayout(delay_lay, 4, 2)
+
+        return lay
+
+    def _graphsLayout(self):
+        lay = QVBoxLayout()
+
+        self.amplitude_graph = self.setupPlot()
+        self.amplitude_graph.setObjectName('amplitude_graph')
+        self.amplitude_graph.setStyleSheet(
+            '#amplitude_graph{min-height:15em;min-width:21em;max-height:15em;}')
+
+        self.phase_graph = self.setupPlot()
+        self.phase_graph.setObjectName('phase_graph')
+        self.phase_graph.setStyleSheet(
+            '#phase_graph{min-height:15em;min-width:21em;max-height:15em;}')
+
+        checks_wid = QWidget()
+        self.hbox_checks = QHBoxLayout(checks_wid)
+        for idx in range(len(self.chs['FDL']['Signals'])):
+            self.setupCurve(
+                self.chs['FDL']['Signals'][idx], 
+                self.prefix + self.chs['FDL']['Time'], 
+                idx)
+        
+        lay.addWidget(checks_wid)
+        lay.addWidget(QLabel('<h4>Amplitude</h4>'))
+        lay.addWidget(self.amplitude_graph)
+        lay.addWidget(QLabel('<h4>Phase</h4>'))
+        lay.addWidget(self.phase_graph)
+
+        return lay
+
+    def setupPlot(self):
+        graph = SiriusWaveformPlot(
+            parent=self, background=QColor(255, 255, 255))
+        graph.maxRedrawRate = 2
+        graph.setShowXGrid(True)
+        graph.setShowYGrid(True)
+        graph.setShowLegend(False)
+        graph.setAutoRangeX(True)
+        graph.setAutoRangeY(True)
+        graph.setAxisColor(QColor(0, 0, 0))
+
+        return graph
+    
+    def setupCurve(self, signal, time, idx):
```

Comment:
avoid using the name "time", because of the python library that has this name
maybe rename to timebase